First stage in instant file search results.

ryecroft opened the pull request:
The initial work for searches for files in the same folder as the
current file to appear in a new buffer and update as you type.
Tito to finish off and add threading.

ryecroft committed:
First stage in instant file search results.

The initial work for searches for files in the same folder as the
current file to appear in a new buffer and update as you type.
Tito to finish off and add threading.

titoBouzout commented on SideBar.py:
```diff
@@ -270,25 +270,48 @@ def description(self, paths = []):
+	
+	# - The big one: I don't understand how to turn on and off the event listener,
+	#		so it is currently on all the time.
+	
```
HI :)  Is not possible to turn it off. That's why we need to check if the view is our own view.

titoBouzout commented on SideBar.py:
```diff
@@ -270,25 +270,48 @@ def description(self, paths = []):
+	
+	# - I couldn't set a sidebar_results variable that I could then access
+	#		from the FindResultsListener, so I went off the name of the view.
+	
```
You can set and get the value via:
`view.settings().set('sidebar_results', True)`
`view.settings().get('sidebar_results')`

ryecroft replied:
Aha, ok. I tried to do:

`view.sidebar_results = True`

which seemed to work, but then wasn't there in the event listener.

On 22/03/2012, at 11:30, tito wrote:

> > -    ##       ##     ##    ##          ##     ##   ##  ##     ##    ##  
> > -    ######   ########      ######     ##    ##     ## ########     ##  
> > -    ##       ##     ##          ##    ##    ######### ##   ##      ##  
> > -    ##       ##     ##    ##    ##    ##    ##     ## ##    ##     ##  
> > -    ##       ########      ######     ##    ##     ## ##     ##    ##  
> > -   
> > -    # Hi Tito!  
> > -   
> > -    # So, I've had a go with it. I had a couple of problems:
> > -   
> > -    # - The big one: I don't understand how to turn on and off the event listener,
> > -    #       so it is currently on all the time.
> > -   
> > -    # - I couldn't set a sidebar_results variable that I could then access
> > -    #       from the FindResultsListener, so I went off the name of the view.
> > -   
> 
> You can set and get the value via:
> `view.settings().set('sidebar_results', True)`
> `view.settings().get('sidebar_results')`
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/titoBouzout/SideBarEnhancements/pull/11/files#r590250

## 

Stephen Horne

titoBouzout replied:
That object seems to be non-writeable. That's why view.settings() is there.

ryecroft replied:
Yo Tito,

Just in case you were going to look at it tomorrow, don't!

I've got it working very nicely with threads now (after three hours of trying), so I will try to finish off the final bits that I am struggling with and  push again tomorrow.

Thanks,

Fb

On 22/03/2012, at 11:42, tito wrote:

> > -    ##       ##     ##    ##          ##     ##   ##  ##     ##    ##  
> > -    ######   ########      ######     ##    ##     ## ########     ##  
> > -    ##       ##     ##          ##    ##    ######### ##   ##      ##  
> > -    ##       ##     ##    ##    ##    ##    ##     ## ##    ##     ##  
> > -    ##       ########      ######     ##    ##     ## ##     ##    ##  
> > -   
> > -    # Hi Tito!  
> > -   
> > -    # So, I've had a go with it. I had a couple of problems:
> > -   
> > -    # - The big one: I don't understand how to turn on and off the event listener,
> > -    #       so it is currently on all the time.
> > -   
> > -    # - I couldn't set a sidebar_results variable that I could then access
> > -    #       from the FindResultsListener, so I went off the name of the view.
> > -   
> 
> That object seems to be non-writeable. That's way view.settings() is there.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/titoBouzout/SideBarEnhancements/pull/11/files#r590303

## 

Stephen Horne

ryecroft replied:
Ok, it's done. It will just need some polishing from you.

I think it's a very nice thing. One of the things I don't like about the default GoToFile search is the way it will only search unopened filenames if you have opened a project. Our one fixes that **and** we will be able to open all the matches in one go. Brilliant!

Thanks for the help and encouragement.

Fb

On 22/03/2012, at 11:42, tito wrote:

> > -    ##       ##     ##    ##          ##     ##   ##  ##     ##    ##  
> > -    ######   ########      ######     ##    ##     ## ########     ##  
> > -    ##       ##     ##          ##    ##    ######### ##   ##      ##  
> > -    ##       ##     ##    ##    ##    ##    ##     ## ##    ##     ##  
> > -    ##       ########      ######     ##    ##     ## ##     ##    ##  
> > -   
> > -    # Hi Tito!  
> > -   
> > -    # So, I've had a go with it. I had a couple of problems:
> > -   
> > -    # - The big one: I don't understand how to turn on and off the event listener,
> > -    #       so it is currently on all the time.
> > -   
> > -    # - I couldn't set a sidebar_results variable that I could then access
> > -    #       from the FindResultsListener, so I went off the name of the view.
> > -   
> 
> That object seems to be non-writeable. That's way view.settings() is there.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/titoBouzout/SideBarEnhancements/pull/11/files#r590303

## 

Stephen Horne

titoBouzout replied:
Great FB, good to see you motivated with this.
Just a comment, it seems some config about whitespaces or something changed and now the commit is unreadable. 
https://github.com/titoBouzout/SideBarEnhancements/pull/11/files
Can you please clean the file to keep only relevant changes? Thanks!

ryecroft replied:
> Can you please  clean the commit to keep only relevant changes? Thanks!

Whoops. I had converted whitespace to spaces, but they were tabs before. I have changed it back to tabs.

## 

Fb

titoBouzout commented:
Great, I will take a look later today or tomorrow.

Looks promising. If you find something to correct or whatever: pushing again will update this pull request. So don't hesitate and if you wish do it.

Regards,

ryecroft commented:
Ok, sounds good.

Catch you later.

Fb

On 22/03/2012, at 11:36, tito wrote:

> Great, I will take a look later today or tomorrow.
> 
> Looks promising. If you find something to correct or whatever: pushing again will update this pull request. So don't hesitate and if you wish do it.
> 
> Regards,
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/titoBouzout/SideBarEnhancements/pull/11#issuecomment-4636581

## 

Stephen Horne

ryecroft committed:
Alt-f interactive search now threaded.

Added the threading capability to the search. It now runs very well.
Remaining to do is add the open_results function, and maybe some
matched string highlighting in the results.

ryecroft committed:
Search-as-you-type feature now implemented and threaded in
SideBarFindFilesPathContainingCommand.

ryecroft committed:
Search-as-you-type feature now implemented and threaded in
SideBarFindFilesPathContainingCommand.

Added search-term highlighting in the search results.
The search window will close on pressing return with the caret in
the first line.
Needs the function to open the results to be built in. I currently use
a hacked version or open-include, but it should be in the same package.

ryecroft committed:
Search-as-you-type feature now implemented and threaded in
SideBarFindFilesPathContainingCommand.

Added search-term highlighting in the search results.
The search window will close on pressing return with the caret in
the first line.

ryecroft committed:
Search-as-you-type feature now implemented and threaded in
SideBarFindFilesPathContainingCommand.

Added search-term highlighting in the search results.
The search window will close on pressing return with the caret in
the first line.

titoBouzout commented:
Hi FB! I merged your contribution with some changes:
- the original function was allowing to "search" on selected directories of the sidebar. I restored this possibility.
- you mentioned:

```
# I use your open-include, but I had to modify it to make it just use the absolute paths,
# and I had to delete the bit that checked if the filename was None
```

If there is a problem with Open-Include opening files I'm hope we can fixit. I was unable to understand the problem, can you please provide some more details? so I can work on that.
- I simplified the code and remove the "enter catch" hack, I found this to hacky. ALT+D is not that hard to press!
- The heading: "Searching '+str(self.num_files)+' files for "'+self.searchTerm+'" in" is usefull for when we are searching on many folders at the same time.
- the is_enabled method was in the wrong class.
- optimized a little bit the threading code.
- By default will search in all the project folders. I found current folder search being not useful.

Now I remember I forget to highlight matches, will push this in minutes..

Thanks a lot!

titoBouzout commented:
Highlight matches pushed!

ryecroft commented:
Yo Tito, looking good now =]

Very polished.

> - you mentioned:
> 
> ```
> # I use your open-include, but I had to modify it to make it just use the absolute paths,
> # and I had to delete the bit that checked if the filename was None
> ```
> 
> If there is a problem with Open-Include opening files I'm hope we can fixit. I was unable to understand the problem, can you please provide some more details? so I can work on that.

This is due to the way I work. I often open files as files, not as a project. If I open one file from the OS, it will open Sublime and the file. If I then run our new search plugin and end up with a list of, say, 6 files, if I now press alt-d to open_include, it will only open one file, not all 6 (OSX).

> - I simplified the code and remove the "enter catch" hack, I found this to hacky. ALT+D is not that hard to press!

Yes, I agree about the hacky part =\  I preferred return because it's the natural thing to do.

I also thought that the opening functionality would do well to be built in, rather than depend on the installation of another package, but that would create some overlap, which is definitely not cool.

> - The heading: "Searching '+str(self.num_files)+' files for "'+self.searchTerm+'" in" is usefull for when we are searching on many folders at the same time.
> - the is_enabled method was in the wrong class.
> - optimized a little bit the threading code.
> - By default will search in all the project folders. I found current folder search being not useful.

I think that this may be the cause of my troubles I mentioned above, maybe you can put the current folder in as a fallback if there is no project folder?

> Now I remember I forget to highlight matches, will push this in minutes..
> 
> Thanks a lot!

No worries - it's been nice to collaborate on something, thanks for all the help!

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/titoBouzout/SideBarEnhancements/pull/11#issuecomment-4671186

## 

Stephen Horne

titoBouzout commented:
>  and end up with a list of, say, 6 files, if I now press alt-d to open_include, it will only open one file, not all 6 (OSX).

It should open the 6 files, Can you please try this:
Open "open_include.py" and on line 65, press enter and add `print paths` and paste the result here.
Remember that if your cursor is on 1 file, it will open only 1 file. If your cursor is not in a file, will open all files.
( in a file /path/ string )

> I think that this may be the cause of my troubles I mentioned above, maybe you can put the current folder in as a fallback if there is no project folder?

That's already in! Please try it.

Thanks to you.

titoBouzout commented:
OK I noticed the problem with open include. Let me check.

titoBouzout commented:
I fixed "open include"
Regards

ryecroft commented:
Hi Tito,

Sorry - I staggered off to bed last night so I missed your next emails.

It now works beautifully! Thanks.

Fb

On 24/03/2012, at 04:24, tito wrote:

> > and end up with a list of, say, 6 files, if I now press alt-d to open_include, it will only open one file, not all 6 (OSX).
> 
> It should open the 6 files, Can you please try this:
> Open "open_include.py" and on line 65, press enter and add `print paths` and paste the result here.
> Remember that if your cursor is on 1 file, it will open only 1 file. If your cursor is not in a file, will open all files.
> 
> > I think that this may be the cause of my troubles I mentioned above, maybe you can put the current folder in as a fallback if there is no project folder?
> 
> That's already in! Please try it.
> 
> Thanks to you.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/titoBouzout/SideBarEnhancements/pull/11#issuecomment-4671565

## 

Stephen Horne
